### PR TITLE
add create_tag.sh and improve top level docs

### PR DESCRIPTION
**Pull request by nslater**

### CODE REVIEW

File: DEVELOP.rst [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    $ bundle exec rspec spec
+
+Preparing a Release
+===================
+
+To create a new release, you must:
+
+- Update ``CrateRuby.version`` in ``lib/crate_ruby/version.rb``
+
```
Comment (chaudum) [inline DEVELOP.rst]: do lists with newlines between bullet points work in `rst`?

File: devtools/create_tag.sh [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+if [ "$LOCAL_COMMIT" != "$ORIGIN_COMMIT" ]
+then
+   echo "Local $BRANCH is not up to date. "
+   echo "Aborting."
+   exit -1
+fi
+
+# check if tag to create has already been created
+WORKING_DIR=`dirname $0`
+VERSION=`$WORKING_DIR/../bin/py setup.py --version`
```
Comment (chaudum) [inline devtools/create_tag.sh]: this only works for python packages!

File: DEVELOP.rst (hunk 1 above)
Comment (nslater) [inline DEVELOP.rst]: yup. blank lines are optional

File: devtools/create_tag.sh (hunk 2 above)
Comment (nslater) [inline devtools/create_tag.sh]: fixed the script to work with ruby gems! ty

**Pull request merged**